services/server/src/server/services/storageServices/SourcifyDatabaseService.ts

```diff
@@ -1053,4 +1054,59 @@ export class SourcifyDatabaseService
       );
     });
   }
+
+  async getSimilarityCandidatesByRuntimeCode(
+    runtimeBytecode: string,
+    limit: number,
+  ): Promise<GetSourcifyMatchByChainAddressWithPropertiesResult[]> {
+    await this.init();
+
+    const runtimeBuffer = bytesFromString(runtimeBytecode);
+    if (!runtimeBuffer || runtimeBuffer.length === 0) {
+      return [];
```

Comment:
I think we should throw here. If this case is encountered, we have an unexpected server error.